base/applications/cleanmgr/util.c

```diff
@@ -0,0 +1,1179 @@
+        SetDlgItemTextW(hwnd, IDC_STATIC_INFO, L"Temporary files");
+    }
+    
+    if (bv.RecycleClean == TRUE)
+    {   
+        StringCbCopyW(TargetedDir, _countof(TargetedDir), wcv.DriveLetter);
+        StringCbCatW(TargetedDir, _countof(TargetedDir), L"\\");
+        
+        SHEmptyRecycleBinW(NULL, TargetedDir, SHERB_NOCONFIRMATION | SHERB_NOPROGRESSUI | SHERB_NOSOUND);
+        
+        SendMessageW(hProgressBar, PBM_SETPOS, 50, 0);
+        SetDlgItemTextW(hwnd, IDC_STATIC_INFO, L"Recycled files");
+    }
+    
+    if (bv.ChkDskClean == TRUE)
+    {
+        for(int i = 0; i < 10; i++)
+        {
+            StringCchPrintfW(TargetedDir, _countof(TargetedDir), L"%s\\FOUND.%.3i", wcv.DriveLetter, i);
+            if (PathIsDirectoryW(TargetedDir))
+            {
+                CleanRequiredPath(TargetedDir);
+            }
+        
+            else
+                break;
+        }
+        SendMessageW(hProgressBar, PBM_SETPOS, 75, 0);
+        SetDlgItemTextW(hwnd, IDC_STATIC_INFO, L"Old ChkDsk files");
+    }
+
+    if (bv.RappsClean == TRUE)
+    {
+        CleanRequiredPath(wcv.RappsDir);
+
+        SendMessageW(hProgressBar, PBM_SETPOS, 100, 0);
+        SetDlgItemTextW(hwnd, IDC_STATIC_INFO, L"Temporary RAPPS Files");
+    }
+    
+    SendMessageW(hwnd, WM_DESTROY, 0, 0);
+
+    return TRUE;
+}
+
+void CleanRequiredPath(PWCHAR TempPath)
+{
+    SHFILEOPSTRUCTW FileStruct;
+    ZeroMemory(&FileStruct, sizeof(FileStruct));
+
+    int ArrLen = wcslen(TempPath) + 2; // required to set 2 nulls at end of argument to SHFileOperation.
+    WCHAR* Path = (WCHAR*) malloc(ArrLen);
+    memset(Path, 0, ArrLen);
+    StringCbCopyW(Path, ArrLen, TempPath);
+    
+    FileStruct.wFunc = FO_DELETE;
+    FileStruct.fFlags = FOF_SILENT | FOF_NOCONFIRMATION | FOF_NOERRORUI | FOF_NOCONFIRMMKDIR;
+    FileStruct.fAnyOperationsAborted = FALSE;
+    FileStruct.pFrom = TempPath;
+
+    SHFileOperationW(&FileStruct);
+    free(Path);
+}
+
+uint64_t DirSizeFunc(PWCHAR TargetDir)
+{
+    WIN32_FIND_DATAW DataStruct;
+    ZeroMemory(&DataStruct, sizeof(DataStruct));
+    LARGE_INTEGER sz;
+    ZeroMemory(&sz, sizeof(sz));
+    HANDLE HandleDir = NULL;
+    uint64_t Size = 0;
+    WCHAR TargetedDir[2048] = { 0 };
+    WCHAR ReTargetDir[2048] = { 0 };
+    DWORD RetError = 0;
+    
+    StringCchPrintfW(TargetedDir, _countof(TargetedDir), L"%s%s", TargetDir, L"\\*");
+    HandleDir = FindFirstFileW(TargetedDir, &DataStruct);
+    RetError = GetLastError();
+    
+    if (RetError != ERROR_NO_MORE_FILES && HandleDir == INVALID_HANDLE_VALUE)
+    {
+        if (RetError == ERROR_PATH_NOT_FOUND)
+        {
+            return 0;
+        }
+
+        else
+        {
+            MessageBoxW(NULL, L"FindFirstFileW() failed!", L"Error", MB_OK | MB_ICONERROR);
+            return 1;
+        }
+    }
+
+    do
+    {
+        if (wcscmp(DataStruct.cFileName, L".") != 0 && wcscmp(DataStruct.cFileName, L"..") != 0)
```

Comment:
```suggestion
        if (wcscmp(DataStruct.cFileName, L".") == 0 || wcscmp(DataStruct.cFileName, L"..") == 0)
            continue;
```
and you could save one level of indentation for the code below.